go/exocom/test_helpers/mock_service.go

```diff
@@ -69,10 +71,9 @@ func (m *MockService) Close() error {
 
 //GetReceivedMessages gets the messages this mock service has received
 func (m *MockService) GetReceivedMessages() []structs.Message {
-	if m.ReceivedMessages == nil {
-		return []structs.Message{}
-	}
-	return m.ReceivedMessages
+	m.receivedMessagesMutex.RLock()
+	defer m.receivedMessagesMutex.RUnlock()
+	return m.receivedMessages
```

Comment:
I'd say if an update of this variable happens in parallel to reading it, returning an older value is not necessarily wrong behavior. Or can the value we read here be garbage while being updated somewhere else?